src/betterdiscord/ui/notifications.tsx

```diff
@@ -1,131 +1,136 @@
 import React, {ReactDOM} from "@modules/react";
-import Button from "@ui/base/button";
+import Button, {type ButtonProps, Colors, Looks} from "@ui/base/button";
 import Settings from "@stores/settings";
+import Notifications from "@stores/notifications";
 import Text from "@ui/base/text";
-import {CircleAlertIcon, InfoIcon, TriangleAlertIcon, CircleCheckIcon} from "lucide-react";
+import {CircleAlertIcon, CircleCheckIcon, InfoIcon, TriangleAlertIcon} from "lucide-react";
 import DOMManager from "@modules/dommanager";
 import DiscordModules from "@modules/discordmodules";
-import type {MouseEvent} from "react";
-import type {Position} from "./settings/components/position";
+import type {MouseEvent, ReactNode} from "react";
+import {useInternalStore} from "@ui/hooks.ts";
+import {shallowEqual} from "fast-equals";
+import Markdown from "@ui/base/markdown.tsx";
+import ErrorBoundary from "@ui/errorboundary.tsx";
 
+const spring = DiscordModules.ReactSpring;
 
 // TODO: let arven fix this
 export type NotificationType = "warning" | "error" | "info" | "success";
-export interface NotificationAction {
+
+interface ButtonActions extends ButtonProps {
     label: string;
-    onClick?(): void;
+    dontClose?: boolean;
+    dontCloseOnActionIfHoldingShiftKey: boolean;
 }
+
 export interface Notification {
     id: string;
     title?: string;
-    content?: string;
+    content?: string | ReactNode;
     type?: NotificationType;
     duration?: number;
-    actions: NotificationAction[];
-    onDurationDone?(): void;
+    actions: ButtonActions[];
+
+    onClose?(): void;
+
     onClick?(): void;
-    exiting?: boolean;
+
     icon?: React.FC;
 }
 
-const Icon = ({type}: {type: NotificationType;}) => {
+const positions: {
+    [x: string]: { top?: number, right?: number, bottom?: number, left?: number, flexDirection: string }
+} = {
+    "top-right": {top: 16, right: 16, flexDirection: "column"},
+    "top-left": {top: 16, left: 16, flexDirection: "column"},
+    "bottom-right": {bottom: 16, right: 16, flexDirection: "column-reverse"},
+    "bottom-left": {bottom: 16, left: 16, flexDirection: "column-reverse"}
+} as const;
```

Comment:
This is typed wrong. The type you seem to want is
```suggestion
const positions = {
    "top-right": {top: 16, right: 16, flexDirection: "column"},
    "top-left": {top: 16, left: 16, flexDirection: "column"},
    "bottom-right": {bottom: 16, right: 16, flexDirection: "column-reverse"},
    "bottom-left": {bottom: 16, left: 16, flexDirection: "column-reverse"}
} satisfies {
    [x: string]: { top?: number, right?: number, bottom?: number, left?: number, flexDirection: string }
};
```